### PR TITLE
Set scoreAndGrade's selected grade candidate and use it when saving a…

### PR DESCRIPTION
**Pull request by wongvincent**
…n activityUsage

### CODE REVIEW

File: components/d2l-activity-editor/state/activity-score-grade.js [hunk 1]
```diff
@@ -37,6 +40,8 @@ export class ActivityScoreGrade {
 
 	removeFromGrades() {
 		this.inGrades = false;
+		this.associatedGrade = null;
```
Comment (mdgbayly) [inline components/d2l-activity-editor/state/activity-score-grade.js]: Could update tests in `activity-score-grade.spec.js` to account for this.

File: components/d2l-activity-editor/state/activity-score-grade.js [hunk 2]
```diff
@@ -60,6 +65,15 @@ export class ActivityScoreGrade {
 		}
 		return !this.scoreOutOfError;
 	}
+
+	setAssociatedGrade(gradeCandidate) {
```
Comment (mdgbayly) [inline components/d2l-activity-editor/state/activity-score-grade.js]: Could update tests in `activity-score-grade.spec.js` to account for this.

**Comment by wongvincent** [comment]
> One change that might make things a little clearer is to give the ActivityScoreGrade.associatedGrade a more descriptive name, as I think it only has a value when the associated grade is changing?

It won't have a value if it is currently associated to something and we don't ever change it to associate to any other grade. If we change it to associate to something else, then back to the original association, then it will have a value.

File: components/d2l-activity-editor/state/activity-score-grade.js (hunk 1 above)
Comment (wongvincent) [reply components/d2l-activity-editor/state/activity-score-grade.js]: Today will be a day of writing unit tests. Including for the MobX grade objects I added!

**Comment by wongvincent** [comment]
This visual diff was failing: https://s3.ca-central-1.amazonaws.com/visualdiff.gaudi.d2l/screenshots/BrightspaceHypermediaComponents/activities/action-dismiss-dialog/2020-03-11-06.50.52.750-602ef8483b3d171cb6b643acc2457b3784114508-report.html

File: components/d2l-activity-editor/state/activity-score-grade.js (hunk 2 above)
Comment (mdgbayly) [inline components/d2l-activity-editor/state/activity-score-grade.js]: nit: I think we are still missing a test for this action?

Comment (wongvincent) [reply components/d2l-activity-editor/state/activity-score-grade.js]: next pr :)

Comment (wongvincent) [reply components/d2l-activity-editor/state/activity-score-grade.js]: https://github.com/BrightspaceHypermediaComponents/activities/pull/763/commits/c8be395cdb56525286c6dee5e504e0867562922d

**Pull request merged**